tests/functional/pv/pvc_clone/test_clone_deletion_after_max_cluster_space_utilization.py

```diff
@@ -0,0 +1,213 @@
+                    f"The actaul alerts: {actual_alerts}"
+                )
+                return False
+        return True
+
+    @skipif_external_mode
+    @magenta_squad
+    def test_clone_deletion_after_max_cluster_space_utilization(
+        self, interface_type, pvc_clone_factory, threading_lock
+    ):
+        """
+        Steps:
+            1. Have a cluster with OCP + ODF internal mode
+            2. Create cephfs + RBD pvcs with data
+            3. Create clones of pvcs till cluster full ratio.
+            4. Make the cluster out of full by increasing the full ratio.
+            5. After the cluster is out of full state and IOs started , Try to delete clones.
+            6. Clone deletion should be successful and should not give error messages.
+        """
+        # Creating the clones one by one and wait until they bound
+        self.timeout = 1800
+        logger.info(
+            f"Start creating {self.num_of_clones} clones on {interface_type} PVC of size {self.pvc_size} GB."
+        )
+        clones_list = []
+        for i in range(self.num_of_clones):
+            index = i + 1
+            logger.info(f"Start creation of clone number {index}.")
+
+            cloned_pvc_obj = pvc_clone_factory(
+                self.pvc_obj,
+                storageclass=self.pvc_obj.backed_sc,
+            )
+
+            helpers.wait_for_resource_state(
+                cloned_pvc_obj, constants.STATUS_BOUND, timeout=1800
+            )
+            cloned_pvc_obj.reload()
+            clones_list.append(cloned_pvc_obj)
+            logger.info(
+                f"Clone with name {cloned_pvc_obj.name} for {self.pvc_size} pvc {self.pvc_obj.name} was created."
+            )
+
+        logger.info("Verify Alerts are seen 'CephClusterErrorState'")
+
+        expected_alerts = ["CephClusterErrorState"]
+        sample = TimeoutSampler(
+            timeout=600,
+            sleep=10,
+            func=self.verify_alerts_via_prometheus,
+            expected_alerts=expected_alerts,
+            threading_lock=threading_lock,
+        )
+        if not sample.wait_for_func_status(result=True):
+            logger.error(f"The alerts {expected_alerts} do not exist after 1200 sec")
+            raise TimeoutExpiredError
+
+        # Make the cluster out of full by increasing the full ratio.
+        logger.info("Change Ceph full_ratio from from 20% to 50%")
+
+        change_ceph_full_ratio(50)
+        # After the cluster is out of full state and IOs started , Try to delete clones.
+        # Delete the clones one by one and wait for deletion
+        logger.info(
+            f"Start deleting {self.num_of_clones} clones on {interface_type} PVC of size {self.pvc_size} Gi."
+        )
+        index = 0
+        for clone in clones_list:
+            index += 1
+            pvc_reclaim_policy = clone.reclaim_policy
+            clone.delete()
+            logger.info(
```

Comment:
is it important to mention the clone number? just clone name is not enough?

Reply:
either way is fine

Reply:
yeah if its not vital to the test flow, you can simply remove it.

Reply:
Removed clone number